### PR TITLE
downloadjs types added

### PR DESCRIPTION
**Pull request by cwmoo740**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

 https://github.com/rndme/download

### CODE REVIEW

File: types/downloadjs/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// TypeScript Version: 2.2
+
+declare namespace download {
+    type Data = string | File | Blob;
+    interface Stringable {
+        toString(): string;
+    }
+}
+declare function download(data: download.Data, filename?: string, mimeType?: string): void;
+declare function download<T extends download.Stringable>(data: T, filename?: string, mimeType?: string): void;
```
Comment (andy-ms) [inline types/downloadjs/index.d.ts]: `toString` is defined on `Object`, so almost everything is Stringable. If it's just used for the string value, it's better to just require that a string be passed in. Just like `parseInt({ toString: () => "123" });` is valid JavaScript but TypeScript requires an actual string there.

File: types/downloadjs/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Type definitions for downloadjs 1.4
+// Project: http://danml.com/download.html
+// Definitions by: cwmoo740 <https://github.com/cwmoo740>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+declare namespace download {
+    type Data = string | File | Blob;
```
Comment (andy-ms) [inline types/downloadjs/index.d.ts]: Only used in one place, inline it.

File: types/downloadjs/index.d.ts (hunk 1 above)
Comment (cwmoo740) [inline types/downloadjs/index.d.ts]: The library provides this example:

```js
var str= "hello world",	arr= new Uint8Array(str.length);
str.split("").forEach(function(a,b){
  arr[b]=a.charCodeAt();
});

download( arr, "textUInt8Array.txt", "text/plain" );
```
I'm not sure why anyone would want to do that, but given that the library calls toString() on whatever is passed as the first argument, I thought I would add the type for that. 

I'm happy removing it too.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**